campaignion_layout/tests/ThemesTest.php

```diff
@@ -55,23 +63,75 @@ public function testEnabledThemes() {
   }
 
   /**
-   * Test getting the declared layouts.
+   * Create themes instance with a list of themes.
    */
-  public function testDeclaredLayouts() {
-    $foo = $this->createMock(Theme::class);
-    $foo->method('invokeLayoutHook')->willReturn([
-      'foo' => ['title' => 'Foo'],
-    ]);
-    $bar = $this->createMock(Theme::class);
-    $bar->method('invokeLayoutHook')->willReturn([
-      'bar' => ['title' => 'Bar'],
-    ]);
+  protected function getThemes($mock_themes) {
     $cache = $this->createMock(\DrupalCacheInterface::class);
     $themes = $this->getMockBuilder(Themes::class)
       ->setConstructorArgs([[], $cache])
-      ->setMethods(['enabledThemes'])
+      ->setMethods(['allThemes'])
       ->getMock();
-    $themes->method('enabledThemes')->willReturn([$foo, $bar]);
+    $themes->method('allThemes')->willReturn($mock_themes);
+    return $themes;
+  }
+
+  /**
+   * Create a mock theme.
+   */
+  protected function mockTheme($bases = [], $enabled = TRUE, $feature = TRUE, $layouts = [], $invoke = TRUE) {
+    $theme = $this->createMock(Theme::class);
+    $theme->method('isEnabled')->willReturn($enabled);
+    $theme->method('hasFeature')->willReturn($feature);
+    $theme->method('baseThemes')->willReturn($bases);
+    $theme->expects(isset($layouts) ? $this->once() : $this->never())
+      ->method('invokeLayoutHook')->willReturn($layouts);
+    return $theme;
+  }
+
+  /**
+   * Test getting the declared layouts with a disabled base theme.
+   */
+  public function testDeclaredLayoutsDisabledBase() {
+    $theme['foo_base'] = $this->mockTheme([], TRUE, TRUE, []);
```

Comment:
The test name says "disabled base" but with `TRUE` as second argument it is not actually disabled, is it? It just doesn’t declare any layouts?

Reply:
Fixed, thanks. It still checks whether the hook was invoked on the base theme though.